src/provider/application.js

```diff
@@ -9,17 +9,29 @@ import MutationObserver from 'mutation-observer';
 
 /** Application class which represents an embedded application. */
 class Application extends EventEmitter {
-  init({ acls = [], secret = null, onReady = null }) {
+  init({ acls = [], secret = null, onReady = null, autoAuthorizeConsumer = false }) {
```

Comment:
Can we just have this variable called ‘authorize’ with a default set to true

Reply:
Actually would this be simpler if ACLs is just “*” to auto authorize?  Then we don’t need the extra config

Reply:
So when user sets `acls` to be "*" and keeps `secret` being null, we consider it auto authorizing?

Reply:
Yeah

Reply:
I had thought about that, but I felt like it would be better to be more explicit.

Reply:
Well if it’s set to * and no secret  then there is really no security at that point so makes no sense to hide content IMO

Reply:
I'm fine with that, I'll go ahead and make those changes.

Reply:
@mhemesath and @kafkahw, I updated the PR, let me know if you have any issue with the new implementation.

Reply:
looks good to me. I'm assuming doc & test changes will come later?